test/persistence.test.js

```diff
@@ -121,6 +225,26 @@ describe('Persistence', function () {
     assert.deepStrictEqual(treatedData[1], { _id: '3', today: now })
   })
 
+  it('If a doc contains $$deleted: true, no error is thrown if the doc wasnt in the list before (with stream)', function (done) {
+    const now = new Date()
+    const rawData = model.serialize({ _id: '1', a: 2, ages: [1, 5, 12] }) + '\n' +
+      model.serialize({ _id: '2', $$deleted: true }) + '\n' +
+      model.serialize({ _id: '3', today: now })
+    const stream = new Readable()
+
+    stream.push(rawData)
+    stream.push(null)
+
+    d.persistence.treatRawStream(stream, function (err, result) {
+      var treatedData = result.data
```

Comment:
We should `assert.isNull(err)` and `var` should be replaced with `const` (standardjs)